.github/workflows/github_actions.yml

```diff
@@ -0,0 +1,100 @@
+name: TileDB-Spark
+
+on:
+  push:
+    tags:
+      - '*'
+    branches:
+      - 'master'
+  pull_request:
+
+jobs:
+  Ubuntu-test:
+#    if: github.ref == 'refs/heads/master'
+    name: Ubuntu-latest
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - run: unset SYSTEM; ./gradlew checkFormat;
+        shell: bash
+        name: checkFormat
+      - run: unset SYSTEM; ./gradlew assemble;
+        shell: bash
+        name: assemble
+      - run: unset SYSTEM; ./gradlew test;
+        name: test
+        shell: bash
+
+  MacOS-test:
+#    if: github.ref == 'refs/heads/master'
+    name: macOS
+    runs-on: macos-10.15
+    steps:
+      - uses: actions/checkout@v2
+      - run: unset SYSTEM; ./gradlew checkFormat;
+        shell: bash
+        name: checkFormat
+      - run: unset SYSTEM; ./gradlew assemble;
+        shell: bash
+        name: assemble
+      - run: unset SYSTEM; ./gradlew test;
+        shell: bash
+        name: test
+
+  Release:
+    if: startsWith(github.ref, 'refs/tags/')
+    needs: [Ubuntu-test, MacOS-test]
+    name: Create-Release
+    runs-on: ubuntu-latest
```

Comment:
Does this need to run on `macos-latest` as well?

Reply:
It does not matter where the jars for the release are created. I chose ubuntu because it is usually the fastest.